games/minimal/mods/testentities/armor.lua

```diff
@@ -0,0 +1,37 @@
+-- Armorball: Test entity for testing armor groups
+-- Rightclick to change armor group
+
+local phasearmor = {
+	[0]={icy=100},
+	[1]={firy=100},
+	[2]={fleshy=100}
+}
+
+minetest.register_entity("testentities:armorball", {
+	hp_max = 20,
```

Comment:
defining initial properties has been deprecated since the early days of 0.4, but never warned about. A warning will be added soon (#9650)

```lua
initial_properties = {
    hp_max = 20
	physical = false,
	collisionbox = {-0.4,-0.4,-0.4, 0.4,0.4,0.4},
	visual = "sprite",
	visual_size = {x=1, y=1},
	textures = {"testentities_armorball.png"},
	spritediv = {x=1, y=3},
	initial_sprite_basepos = {x=0, y=0},
},
```

Also, the docs say that hp_max is a player only object 

```lua
hp_max = 1,
-- For players only. Defaults to `minetest.PLAYER_MAX_HP_DEFAULT`.
```